Userland/Libraries/LibMarkdown/Text.cpp

```diff
@@ -79,13 +79,19 @@ void Text::TextNode::render_to_html(StringBuilder& builder) const
 
 void Text::TextNode::render_for_terminal(StringBuilder& builder) const
 {
-    String text_copy = text;
-    text_copy.replace("\n", " ");
-    builder.append(text_copy);
+    if (text == "\n" || text.is_whitespace()) {
+        builder.append(" ");
+    } else {
+        builder.append(text);
+    }
```

Comment:
This definitely should be squashed into the appropriate commit.

Reply:
this can't really be squashed without breaking things in the LibMarkdown commit, because it depends on the previous commit, which tokenizes whitespace separately.